### PR TITLE
expose optimizer params, log optimizer type and settings for the run

### PR DESCRIPTION
**Pull request by lessw2020**
This PR addresses the TODO in train.py regarding exposing the optimizer params to toml/cmd line configuration. 

The three settings added are weight decay, beta1 and beta2. 
In addition, it now adds a single line of logging to show which optimizer and optimizer settings are being used for the run:

Screenshot 2024-05-26 at 8 46 37 AM

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -89,22 +89,37 @@ def load_state_dict(self, state_dict) -> None:
...
+            model.parameters(),
+            lr=lr,
+            betas=(beta1, beta2),
+            weight_decay=cfg_weight_decay,
+            foreach=True,
         )
     else:
         raise NotImplementedError(f"Optimizer {name} not added.")
 
+    logger.info(
```
Comment (tianyu-l) [inline train.py:119]: IMO if this is not something people would change often, we probably don't need to print it.

File: train.py [hunk 2]
```diff
@@ -89,22 +89,37 @@ def load_state_dict(self, state_dict) -> None:
...
 
 
-def build_optimizer(model, job_config: JobConfig):
+def build_optimizer(model, job_config: JobConfig, color):
     # build optimizer
     name = job_config.optimizer.name
     lr = job_config.optimizer.lr
+    beta1 = job_config.optimizer.beta1
+    beta2 = job_config.optimizer.beta2
+    cfg_weight_decay = job_config.optimizer.weight_decay
```
Comment (tianyu-l) [inline train.py:98]: we can just call it `weight_decay`

**Review by tianyu-l — COMMENTED** [review]
Changes themselves look OK to me. It's a bit unclear to me how often people would need to tune these parameters. My feeling is that compared with learning rate, betas and weight_decay might have some golden values (the ones we set manually here) and would be quite stable. If that's the case, I'm leaning towards not adding them to configs.

File: torchtitan/config_manager.py [hunk 3]
```diff
@@ -159,6 +159,18 @@ def __init__(self):
         self.parser.add_argument(
             "--optimizer.lr", type=float, default=8e-4, help="Learning rate to use"
         )
+        self.parser.add_argument(
+            "--optimizer.beta1", type=float, default=0.9, help="beta1 rate to use"
+        )
+        self.parser.add_argument(
```
Comment (wanchaol) [inline torchtitan/config_manager.py:165]: beta can just be a tuple which could make it a single cmd arg.

File: torchtitan/config_manager.py [hunk 4]
```diff
@@ -159,6 +159,18 @@ def __init__(self):
...
+            "--optimizer.beta2", type=float, default=0.95, help="beta2 rate to use"
+        )
+        self.parser.add_argument(
+            "--optimizer.weight_decay",
```
Comment (wanchaol) [inline torchtitan/config_manager.py:169]: is `weight_decay` common for other optimizers? I am a little worried that these configs are not generalizable to other types of optimizers. So we might want to think of how to expose the optimizer configs in a more general way, i.e. suppose later we add another optimizer and all of the configs here might not be valid for that optimizer.

**Review by wanchaol — COMMENTED** [review]
I'll second @tianyu-l here. We should only try to expose those cmd args when we want to, or we feel it's necessary or easier to tune the adam optimizers, do you have some motivation to expose this configs?
